codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionRoundtripTest.java

```diff
@@ -348,6 +348,95 @@ public void operationComplete(ChannelFuture future) throws Exception {
     }
 
     @Test
+    public void testSettingsAckIsSentBeforeSettingsAreUsed() throws Exception {
```

Comment:
This doesn't actually check "before settings are used" in the general sense. It only tests `INITIAL_WINDOW_SIZE` handling. Maybe `testSettingsAckIsSentBeforeUsingFlowControl`?

I say that because if the default flow controller avoids calling `writePendingBytes();` from `initialWindowSize(int)`, this test passes. But that wouldn't imply that other settings (e.g., `MAX_CONCURRENT_STREAMS`) would be handled correctly.